pages/api/user/index.ts

```diff
@@ -0,0 +1,26 @@
+import prisma from "@lib/prisma";
+import { NextApiRequest, NextApiResponse } from "next";
+import { getSession } from "next-auth/client";
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const session = await getSession({ req: req });
+  if (!session) {
+    return res.status(401).json({ message: "Not authenticated" });
+  }
+
+  if (req.method === "GET") {
+    const user = await prisma.user.findFirst({
+      where: {
+        id: session.user.id,
+      },
```

Comment:
As far as I see it, we only need the username in this case. With your current code, you're also delivering the password hash and all other user fields to the frontend. This is not neccessarily a security issue (since you have to be logged in, right), but surely some overhead in terms of data amount to be transferred. Since this route is called nearly every time, we should add a select query only requesting the necessary fields.
```suggestion
      },
      select: {
        username: true,
      },
```